AEPCore/Sources/rules/LaunchRulesEngine.swift

```diff
@@ -406,12 +413,23 @@ public class LaunchRulesEngine {
                     Log.trace(label: LOG_TAG, "\(logPrefix) - Event already exists in history, skipping '\(operation)' operation")
                     return
                 }
+                eventExistsInEventHistory = false
+            }
+            let timeoutResult = semaphore.wait(timeout: .now() + .seconds(LaunchRulesEngine.EVENT_HISTORY_LOOKUP_TIMEOUT_SEC))
```

Comment:
We discussed offline yesterday about potentially removing this time out and letting the lookup run since its happening on a separate rules engine thread. Are we planning to keep it or remove it?